requirements.txt

```diff
@@ -3,17 +3,17 @@ setuptools>34.3.0
 wheel
 twine
 flake8
-nose
 nose-progressive
-coverage
-mock
 pytz
-mypy==0.770
-lxml # needed by mypy HTML coverage reporting
 pylint
```

Comment:
The rest of the stuff here is:

1. things we need on the command-line but never in CI (flake8, nose-progressive)
2. stuff we need at package publishing time - maybe we could add another extra fo rthose?  (wheel, twine)
3. cruft that's obsolete or assuredly covered by something else given integration tests pass and these don't get included transitively (pytz, pylint)
4. numpy-stubs (see note below)